[2023-03-20T14:57:44.538+0000] {taskinstance.py:1084} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: project_spark_processing.process_data scheduled__2023-03-19T00:00:00+00:00 [queued]>
[2023-03-20T14:57:44.552+0000] {taskinstance.py:1084} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: project_spark_processing.process_data scheduled__2023-03-19T00:00:00+00:00 [queued]>
[2023-03-20T14:57:44.553+0000] {taskinstance.py:1282} INFO - 
--------------------------------------------------------------------------------
[2023-03-20T14:57:44.553+0000] {taskinstance.py:1283} INFO - Starting attempt 8 of 9
[2023-03-20T14:57:44.553+0000] {taskinstance.py:1284} INFO - 
--------------------------------------------------------------------------------
[2023-03-20T14:57:44.574+0000] {taskinstance.py:1303} INFO - Executing <Task(DataprocSubmitJobOperator): process_data> on 2023-03-19 00:00:00+00:00
[2023-03-20T14:57:44.583+0000] {standard_task_runner.py:55} INFO - Started process 3434 to run task
[2023-03-20T14:57:44.590+0000] {standard_task_runner.py:82} INFO - Running: ['***', 'tasks', 'run', 'project_spark_processing', 'process_data', 'scheduled__2023-03-19T00:00:00+00:00', '--job-id', '200', '--raw', '--subdir', 'DAGS_FOLDER/p_submit_job_draft.py', '--cfg-path', '/tmp/tmpml61sc5y']
[2023-03-20T14:57:44.593+0000] {standard_task_runner.py:83} INFO - Job 200: Subtask process_data
[2023-03-20T14:57:44.624+0000] {warnings.py:110} WARNING - /home/***/.local/lib/python3.7/site-packages/***/settings.py:250: DeprecationWarning: The sql_alchemy_conn option in [core] has been moved to the sql_alchemy_conn option in [database] - the old setting has been used, but please update your config.
  SQL_ALCHEMY_CONN = conf.get("database", "SQL_ALCHEMY_CONN")

[2023-03-20T14:57:44.775+0000] {warnings.py:110} WARNING - /home/***/.local/lib/python3.7/site-packages/***/utils/sqlalchemy.py:124: DeprecationWarning: The sql_alchemy_conn option in [core] has been moved to the sql_alchemy_conn option in [database] - the old setting has been used, but please update your config.
  return not conf.get("database", "sql_alchemy_conn").startswith("mssql")

[2023-03-20T14:57:44.851+0000] {task_command.py:388} INFO - Running <TaskInstance: project_spark_processing.process_data scheduled__2023-03-19T00:00:00+00:00 [running]> on host 4fd00a82f4e9
[2023-03-20T14:57:45.026+0000] {taskinstance.py:1511} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_OWNER=***
AIRFLOW_CTX_DAG_ID=project_spark_processing
AIRFLOW_CTX_TASK_ID=process_data
AIRFLOW_CTX_EXECUTION_DATE=2023-03-19T00:00:00+00:00
AIRFLOW_CTX_TRY_NUMBER=8
AIRFLOW_CTX_DAG_RUN_ID=scheduled__2023-03-19T00:00:00+00:00
[2023-03-20T14:57:45.028+0000] {dataproc.py:1985} INFO - Submitting job
[2023-03-20T14:57:45.030+0000] {crypto.py:83} WARNING - empty cryptography key - values will not be stored encrypted.
[2023-03-20T14:57:45.031+0000] {base.py:73} INFO - Using connection ID 'google_cloud_default' for task execution.
[2023-03-20T14:57:45.294+0000] {dataproc.py:1997} INFO - Job your_job_id2 submitted successfully.
[2023-03-20T14:57:45.327+0000] {dataproc.py:2017} INFO - Waiting for job your_job_id2 to complete
[2023-03-20T14:58:15.625+0000] {taskinstance.py:1775} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/google/cloud/operators/dataproc.py", line 2019, in execute
    job_id=new_job_id, region=self.region, project_id=self.project_id, timeout=self.wait_timeout
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/google/common/hooks/base_google.py", line 468, in inner_wrapper
    return func(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/google/cloud/hooks/dataproc.py", line 723, in wait_for_job
    raise AirflowException(f"Job failed:\n{job}")
airflow.exceptions.AirflowException: Job failed:
reference {
  project_id: "dtc-de-course-375420"
  job_id: "your_job_id2"
}
placement {
  cluster_name: "my-dataproc-cluster"
  cluster_uuid: "e079d383-e991-4d26-a3a2-0b675dd91d2d"
}
pyspark_job {
  main_python_file_uri: "gs://dtc_data_lake_dtc-de-course-375420/spark_job.py"
  args: "--input_table"
  args: "dtc-de-course-375420.chicago_crime_all.et_chicago_crimes_lite"
}
status {
  state: ERROR
  details: "Google Cloud Dataproc Agent reports job failure. If logs are available, they can be found at:\nhttps://console.cloud.google.com/dataproc/jobs/your_job_id2?project=dtc-de-course-375420&region=europe-west6\ngcloud dataproc jobs wait \'your_job_id2\' --region \'europe-west6\' --project \'dtc-de-course-375420\'\nhttps://console.cloud.google.com/storage/browser/dataproc-staging-europe-west6-192108012555-yegdrxcc/google-cloud-dataproc-metainfo/e079d383-e991-4d26-a3a2-0b675dd91d2d/jobs/your_job_id2/\ngs://dataproc-staging-europe-west6-192108012555-yegdrxcc/google-cloud-dataproc-metainfo/e079d383-e991-4d26-a3a2-0b675dd91d2d/jobs/your_job_id2/driveroutput"
  state_start_time {
    seconds: 1679324292
    nanos: 436454000
  }
}
yarn_applications {
  name: "BigQuery to BigQuery with PySpark"
  state: FINISHED
  progress: 1.0
  tracking_url: "http://my-dataproc-cluster-m:8088/proxy/application_1679324038206_0001/"
}
status_history {
  state: PENDING
  state_start_time {
    seconds: 1679324265
    nanos: 277040000
  }
}
status_history {
  state: SETUP_DONE
  state_start_time {
    seconds: 1679324265
    nanos: 314100000
  }
}
status_history {
  state: RUNNING
  state_start_time {
    seconds: 1679324265
    nanos: 772523000
  }
}
driver_control_files_uri: "gs://dataproc-staging-europe-west6-192108012555-yegdrxcc/google-cloud-dataproc-metainfo/e079d383-e991-4d26-a3a2-0b675dd91d2d/jobs/your_job_id2/"
driver_output_resource_uri: "gs://dataproc-staging-europe-west6-192108012555-yegdrxcc/google-cloud-dataproc-metainfo/e079d383-e991-4d26-a3a2-0b675dd91d2d/jobs/your_job_id2/driveroutput"
job_uuid: "aa950da2-3a88-41b0-8284-1c591a6d38ac"
done: true

[2023-03-20T14:58:15.638+0000] {taskinstance.py:1326} INFO - Marking task as UP_FOR_RETRY. dag_id=project_spark_processing, task_id=process_data, execution_date=20230319T000000, start_date=20230320T145744, end_date=20230320T145815
[2023-03-20T14:58:15.667+0000] {standard_task_runner.py:105} ERROR - Failed to execute job 200 for task process_data (Job failed:
reference {
  project_id: "dtc-de-course-375420"
  job_id: "your_job_id2"
}
placement {
  cluster_name: "my-dataproc-cluster"
  cluster_uuid: "e079d383-e991-4d26-a3a2-0b675dd91d2d"
}
pyspark_job {
  main_python_file_uri: "gs://dtc_data_lake_dtc-de-course-375420/spark_job.py"
  args: "--input_table"
  args: "dtc-de-course-375420.chicago_crime_all.et_chicago_crimes_lite"
}
status {
  state: ERROR
  details: "Google Cloud Dataproc Agent reports job failure. If logs are available, they can be found at:\nhttps://console.cloud.google.com/dataproc/jobs/your_job_id2?project=dtc-de-course-375420&region=europe-west6\ngcloud dataproc jobs wait \'your_job_id2\' --region \'europe-west6\' --project \'dtc-de-course-375420\'\nhttps://console.cloud.google.com/storage/browser/dataproc-staging-europe-west6-192108012555-yegdrxcc/google-cloud-dataproc-metainfo/e079d383-e991-4d26-a3a2-0b675dd91d2d/jobs/your_job_id2/\ngs://dataproc-staging-europe-west6-192108012555-yegdrxcc/google-cloud-dataproc-metainfo/e079d383-e991-4d26-a3a2-0b675dd91d2d/jobs/your_job_id2/driveroutput"
  state_start_time {
    seconds: 1679324292
    nanos: 436454000
  }
}
yarn_applications {
  name: "BigQuery to BigQuery with PySpark"
  state: FINISHED
  progress: 1.0
  tracking_url: "http://my-dataproc-cluster-m:8088/proxy/application_1679324038206_0001/"
}
status_history {
  state: PENDING
  state_start_time {
    seconds: 1679324265
    nanos: 277040000
  }
}
status_history {
  state: SETUP_DONE
  state_start_time {
    seconds: 1679324265
    nanos: 314100000
  }
}
status_history {
  state: RUNNING
  state_start_time {
    seconds: 1679324265
    nanos: 772523000
  }
}
driver_control_files_uri: "gs://dataproc-staging-europe-west6-192108012555-yegdrxcc/google-cloud-dataproc-metainfo/e079d383-e991-4d26-a3a2-0b675dd91d2d/jobs/your_job_id2/"
driver_output_resource_uri: "gs://dataproc-staging-europe-west6-192108012555-yegdrxcc/google-cloud-dataproc-metainfo/e079d383-e991-4d26-a3a2-0b675dd91d2d/jobs/your_job_id2/driveroutput"
job_uuid: "aa950da2-3a88-41b0-8284-1c591a6d38ac"
done: true
; 3434)
[2023-03-20T14:58:15.698+0000] {local_task_job.py:212} INFO - Task exited with return code 1
[2023-03-20T14:58:15.735+0000] {taskinstance.py:2585} INFO - 0 downstream tasks scheduled from follow-on schedule check
